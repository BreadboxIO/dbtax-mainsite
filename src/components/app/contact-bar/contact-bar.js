import { ADDRESS } from 'constants/address';
import { Icon } from 'components/layout/icon/icon';
import { PHONE_NUMBER } from 'constants/phone-number';
import React from 'react';
import { convertPhoneNumberToTelLink } from 'util/convert-phone-number-to-tel-link/convert-phone-number-to-tel-link';
import styles from './contact-bar.style.scss';

const phoneLink = convertPhoneNumberToTelLink(PHONE_NUMBER);

export const ContactBar = () => (
  <div className={styles.root}>
    <div className="max-width--xl margin--horizontal--auto">
      <a className="animated fadeIn" href={phoneLink}>
        <Icon name="phone-alt" /> {PHONE_NUMBER}
      </a>
      <a
        className="animated fadeIn margin--left--half"
        href={ADDRESS.GOOGLE_MAPS}
        rel="noopener noreferrer"
        target="_blank"
      >
        <Icon name="home" /> {ADDRESS.STREET}
        <span className="display--xs--none display--sm--inline-block">
          {' '}
          · {ADDRESS.CITY}, {ADDRESS.STATE} · {ADDRESS.ZIP}
        </span>
      </a>
    </div>
  </div>
);
