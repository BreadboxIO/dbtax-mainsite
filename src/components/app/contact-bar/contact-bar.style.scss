@import '../../../styles/config/variables';
@import '../../../styles/mixins/font-size';
@import '../../../styles/mixins/line-height';

.root {
  @include font-size(0.8);
  @include line-height(0.8);

  background: $color-quaternary;
  color: $color-white;
  padding: $spacer / 2 $spacer;

  a {
    animation-delay: $duration-base;
    color: $color-white;
    text-decoration: none;

    &:hover {
      color: $color-secondary;
    }

    &:nth-child(2) {
      animation-delay: $duration-base * 2;
    }
  }
}
