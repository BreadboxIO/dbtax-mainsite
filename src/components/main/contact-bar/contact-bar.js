import { ADDRESS } from 'constants/address';
import { Icon } from 'components/icon/icon';
import { PHONE } from 'constants/phone-number';
import React from 'react';
import styles from './contact-bar.style.scss';

export const ContactBar = () => (
  <div className={styles.root}>
    <div className="max-width--xl margin--horizontal--auto">
      <a className="animated fadeIn" href={`tel:+${PHONE.CALL}`} target="_blank" rel="noopener noreferrer">
        <Icon name="phone-alt" /> {PHONE.DISPLAY}
      </a>
      <a
        className="animated fadeIn margin--left--half"
        href={ADDRESS.GOOGLE_MAPS}
        rel="noopener noreferrer"
        target="_blank"
      >
        <Icon name="home" /> {ADDRESS.STREET}
        <span className="display--xs--none display--sm--inline-block">
          {' '}
          · {ADDRESS.CITY}, {ADDRESS.STATE} · {ADDRESS.ZIP}
        </span>
      </a>
    </div>
  </div>
);
