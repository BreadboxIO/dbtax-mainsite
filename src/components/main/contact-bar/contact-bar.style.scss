@import '../../../styles/config/variables';
@import '../../../styles/mixins/font-size.scss';
@import '../../../styles/mixins/line-height.scss';

.root {
  @include font-size(0.8);
  @include line-height(0.8);

  color: $color-white;
  padding: $spacer / 2;
  background: $color-quaternary;

  a {
    color: $color-white;
    text-decoration: none;

    &:hover {
      color: $color-primary;
    }
  }

  .address {
    margin-left: $spacer / 2;
  }

  .city {
    @media screen and (max-width: $width-mobile) {
      display: none;
    }
  }

  .content {
    margin: 0 auto;
    max-width: $width-max;
  }
}
