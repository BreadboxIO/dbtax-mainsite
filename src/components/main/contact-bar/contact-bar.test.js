import { ContactBar } from './contact-bar';
/* global React, create */

describe('<ContactBar />', () => {
  it('should render correctly', () => {
    const component = create(<ContactBar />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
