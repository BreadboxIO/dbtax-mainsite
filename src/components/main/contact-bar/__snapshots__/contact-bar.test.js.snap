// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactBar /> should render correctly 1`] = `
<div
  className="root"
>
  <div
    className="max-width--xl margin--horizontal--auto"
  >
    <a
      className="animated fadeIn"
      href="tel:+14147647730"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fas fa-phone-alt"
      />
       
      414-764-7330
    </a>
    <a
      className="animated fadeIn margin--left--half"
      href="https://www.google.com/maps/place/1125+Milwaukee+Ave,+South+Milwaukee,+WI+53172"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fas fa-home"
      />
       
      1125 Milwaukee Ave.
      <span
        className="display--xs--none display--md--inline-block"
      >
         
        · 
        South Milwaukee
        , 
        WI
      </span>
       
      · 
      53172
    </a>
  </div>
</div>
`;
